pkg/machine/ignition.go

```diff
@@ -423,61 +424,67 @@ func getCerts(certsDir string, isDir bool) []File {
 		files []File
 	)
 
-	certs, err := ioutil.ReadDir(certsDir)
 	if isDir {
-		if err == nil {
-			for _, cert := range certs {
-				b, err := ioutil.ReadFile(filepath.Join(certsDir, cert.Name()))
+		err := filepath.Walk(certsDir, func(path string, info os.FileInfo, err error) error {
+
+			if err == nil && !info.IsDir() {
+				certPath, err := filepath.Rel(certsDir, path)
 				if err != nil {
-					logrus.Warnf("Unable to read cert file %s", err.Error())
-					continue
+					logrus.Warnf("%s", err)
```

Comment:
Why are you creating the file if it might not exists, or you can not read the content, I think the previous code was correct.

Reply:
Warn and continue.

Reply:
This is an error for getting relative path, but you're right - it shouldn't continue.